### PR TITLE
[SPARK-15820][PySpark][SQL]Add Catalog.refreshTable into python API

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

Add Catalog.refreshTable API into python interface for Spark-SQL.
## How was this patch tested?

Existing test.

### CODE REVIEW

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

**Comment by andrewor14** [comment]
Looks good. Jenks retest this please

**Comment by WeichenXu123** [comment]
@andrewor14 It looks strange, I test on my own machine and it is all OK. If it is the test server's problem?

**Comment by MLnick** [comment]
jenkins retest this please

File: python/pyspark/sql/catalog.py [hunk 1]
```diff
@@ -232,6 +232,11 @@ def clearCache(self):
         """Removes all cached tables from the in-memory cache."""
         self._jcatalog.clearCache()
 
+    @since(2.0)
+    def refreshTable(self, tableName):
+        """Invalidate and refresh all the cached the metadata of the given table."""
```
Comment (MLnick) [inline python/pyspark/sql/catalog.py]: There's an extra "the" in this comment, should be `refresh all the cached metadata`

Comment (WeichenXu123) [inline python/pyspark/sql/catalog.py]: @MLnick Done. I also correct the comment with extra "the" in the scala code. Thanks!

**Comment by WeichenXu123** [comment]
@andrewor14 The PR is OK now.

**Comment by WeichenXu123** [comment]
cc @liancheng  Thanks!

**Comment by liancheng** [comment]
LGTM, merging to master and branch-2.0.

**Pull request closed without merging**